mingw-w64-git/git-wrapper.c

```diff
@@ -113,15 +116,52 @@ static void my_path_append(LPWSTR list, LPCWSTR path, size_t alloc)
 	}
 }
 
+bool IsRunningOnARM64()
+{
+	USHORT processMachine = 0;
+	USHORT nativeMachine = 0;
+
+	// Note: IsWow64Process2 is only available in Windows 10 1511+
+	LPFN_ISWOW64PROCESS2 fnIsWow64Process2 = (LPFN_ISWOW64PROCESS2)GetProcAddress(
+		GetModuleHandle(L"kernel32"), "IsWow64Process2");
+
+	if (NULL != fnIsWow64Process2)
+	{
+		if (!fnIsWow64Process2(GetCurrentProcess(), &processMachine, &nativeMachine))
+		{
+			// IsWow64Process2 isn't available
+			return false;
+		}
+	}
+	return nativeMachine == 0xaa64;
+}
+
 static void setup_environment(LPWSTR top_level_path, int full_path)
 {
 	WCHAR msystem[64];
 	LPWSTR path2 = NULL;
 	int len;
 
 	/* Set MSYSTEM */
-	swprintf(msystem, sizeof(msystem),
-		L"MINGW%d", (int) sizeof(void *) * 8);
+	const wchar_t* message;
+
+	if (IsRunningOnARM64()) {
+		message = L"Is ARM64 device\n";
+		swprintf(msystem, sizeof(msystem),
+				L"ARM64");
+	} else {
+		message = L"Not an ARM64 device\n";
+		swprintf(msystem, sizeof(msystem),
+				L"MINGW%d", (int)sizeof(void*) * 8);
+	}
+
+	MessageBox(
```

Comment:
The `MessageBox` is temporary for testing purposes. Will remove this as soon as this PR has been tested by Git for Windows maintainers

Reply:
Unfortunately, I cannot test, as I have no access to any Windows/ARM64 machine. So yes, I'd prefer this diff hunk to be dropped.